code/modules/cm_marines/compound_silo.dm

```diff
@@ -0,0 +1,193 @@
+/obj/structure/compound_silo
+	name = "compound silo"
+	desc = "A large silo containing a compound used to speed up dropship engine cooldowns."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "weldtank_old"
+	density = TRUE
+	anchored = TRUE
+	health = 250
+
+	var/compound_type = COMPOUND_PHI
+
+	var/obj/item/nozzle/attached_to
+	var/atom/tether_holder
+	var/tether_range = 12
+
+/obj/structure/compound_silo/omega
+	compound_type = COMPOUND_OMEGA
+
+/obj/structure/compound_silo/tau
+	compound_type = COMPOUND_TAU
+
+/obj/structure/compound_silo/epsilon
+	compound_type = COMPOUND_EPSILON
+
+/obj/structure/compound_silo/Initialize()
+	. = ..()
+
+	name = "[compound_type] silo"
+	attached_to = new /obj/item/nozzle(src, compound_type)
+	RegisterSignal(attached_to, COMSIG_PARENT_PREQDELETED, .proc/override_delete)
+
+/obj/structure/compound_silo/proc/override_delete()
+	SIGNAL_HANDLER
+	recall_nozzle()
+	return COMPONENT_ABORT_QDEL
+
+/obj/structure/compound_silo/forceMove(atom/dest)
+	recall_nozzle()
+	return ..()
+
+/obj/structure/compound_silo/proc/recall_nozzle()
+	if(ismob(attached_to.loc))
+		var/mob/M = attached_to.loc
+		M.drop_held_item(attached_to)
+
+	attached_to.forceMove(src)
+
+/obj/structure/compound_silo/attack_hand(mob/user)
+	. = ..()
+
+	if(!attached_to || attached_to.loc != src)
+		return
+
+	if(!ishuman(user))
+		return
+
+	playsound(user.loc, 'sound/handling/multitool_pickup.ogg', vol = 15, vary = TRUE)
+	user.put_in_active_hand(attached_to)
+
+/obj/structure/compound_silo/attackby(obj/item/W, mob/user)
+	if(W == attached_to)
+		playsound(user.loc, 'sound/handling/multitool_drop.ogg', vol = 15, vary = TRUE)
+		recall_nozzle()
+		return
+	return ..()
+
+
+/obj/item/nozzle
+	name = "nozzle"
+	icon = 'icons/obj/items/devices.dmi'
+	icon_state = "rcd"
+
+	w_class = SIZE_LARGE
+
+	var/obj/structure/compound_silo/attached_to
+	var/datum/effects/tethering/tether_effect
+
+	var/zlevel_transfer = FALSE
+	var/zlevel_transfer_timer = TIMER_ID_NULL
+	var/zlevel_transfer_timeout = 5 SECONDS
+
+	var/compound_type
+
+/obj/item/nozzle/Initialize(mapload, var/set_compound_type)
+	. = ..()
+	compound_type = set_compound_type
+	if(istype(loc, /obj/structure/compound_silo))
+		attach_to(loc)
+
+/obj/item/nozzle/Destroy()
+	remove_attached()
+	return ..()
+
+/obj/item/nozzle/proc/attach_to(var/obj/structure/compound_silo/to_attach)
+	if(!istype(to_attach))
+		return
+
+	remove_attached()
+
+	attached_to = to_attach
+
+/obj/item/nozzle/proc/remove_attached()
+	attached_to = null
+	reset_tether()
+
+/obj/item/nozzle/proc/reset_tether()
+	SIGNAL_HANDLER
+	if (tether_effect)
+		UnregisterSignal(tether_effect, COMSIG_PARENT_QDELETING)
+		if(!QDESTROYING(tether_effect))
```

Comment:
why use QDESTROYING here? if there's a qdel loop it might be better handled otherwise